server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go

```diff
@@ -266,11 +285,57 @@ func (s *ContentAddressableStorageServer) BatchReadBlobs(ctx context.Context, re
 		} else {
 			blobRsp.Status = &statuspb.Status{Code: int32(codes.OK)}
 		}
+
+		if blobRsp.Status.Code == int32(codes.OK) && s.env.GetConfigurator().GetCacheZstdTranscodingEnabled() && clientAcceptsCompressor(req.AcceptableCompressors, repb.Compressor_ZSTD) {
+			blobRsp = zstdCompressResponse(blobRsp, req.AcceptableCompressors)
```

Comment:
What do you think of putting this under the existing else clause?

Reply:
I was kinda 50/50 on that since it makes the code more nested, but it's probably clearer to put it in the else clause. Done